exports/taskfiles/utils/boost.yaml

```
@@ -0,0 +1,184 @@
+  # @param {string} INSTALL_PREFIX Path prefix of where the project should be installed.
+  # @param {string[]} TARGETS Target libraries to build.
+  # @param {string[]} [EXTRA_ARGS] Any additional arguments to pass to the generate command.
+  generate:
+    internal: true
+    dir: "{{.SOURCE_DIR}}"
+    vars:
+      EXTRA_ARGS:
+        ref: "default (list) .EXTRA_ARGS"
+    requires:
+      vars: ["SOURCE_DIR", "INSTALL_PREFIX", "TARGETS"]
+    cmds:
+      - >-
+        ./bootstrap.sh
+        --prefix="{{.INSTALL_PREFIX}}"
+        --exec-prefix="{{.INSTALL_PREFIX}}"
+        --with-libraries={{(join "," .TARGETS)}}
+        {{- range .EXTRA_ARGS}}
+        "{{.}}"
+        {{- end}}
+
+  # Runs the b2 build step for boost. The caller must have previously called `generate` on
+  # `SOURCE_DIR` for this task to succeed.
+  #
+  # @param {string} SOURCE_DIR Directory containing the boost source.
+  # @param {string[]} [EXTRA_ARGS] Any additional arguments to pass to the build command.
+  # @param {int} [JOBS] The maximum number of concurrent processes to use when building. If
+  # omitted, the b2 default number is used. Before 1.76.0, the number was 1. Since 1.76.0, the
+  # default is the number of cores.
+  build:
+    internal: true
+    dir: "{{.SOURCE_DIR}}"
+    vars:
+      EXTRA_ARGS:
+        ref: "default (list) .EXTRA_ARGS"
+      JOBS: >-
+        {{default "" .JOBS}}
+    requires:
+      vars: ["SOURCE_DIR"]
+    cmds:
+      - >-
+        ./b2
+        {{- range .EXTRA_ARGS}}
+        "{{.}}"
+        {{- end}}
+        {{- if .JOBS}}
+        "-j{{.JOBS}}"
+        {{- end}}
+
+  # Runs the b2 install step for boost. The caller must have previously called `build` on
+  # `SOURCE_DIR` for this task to succeed. If `CMAKE_SETTINGS_DIR` is set, a settings file will be
+  # created in that directory, containing a `boost_ROOT` CMake variable that points to
+  # `INSTALL_PREFIX`.
+  #
+  # @param {string} SOURCE_DIR Directory containing the boost source.
+  # @param {string} INSTALL_PREFIX Path prefix of where the project should be installed.
+  # @param {string} [CMAKE_SETTINGS_DIR] If set, the directory where the project's CMake settings
+  # file should be stored.
+  # @param {string[]} [EXTRA_ARGS] Any additional arguments to pass to the install command.
+  install:
+    internal: true
+    dir: "{{.SOURCE_DIR}}"
+    vars:
+      EXTRA_ARGS:
+        ref: "default (list) .EXTRA_ARGS"
+      INSTALL_PREFIX: >-
+        {{default "" .INSTALL_PREFIX}}
+      CMAKE_SETTINGS_DIR: >-
+        {{default "" .CMAKE_SETTINGS_DIR}}
+    requires:
+      vars: ["SOURCE_DIR", "INSTALL_PREFIX"]
+    cmds:
+      - >-
+        ./b2
+        install
+        {{- range .EXTRA_ARGS}}
+        "{{.}}"
+        {{- end}}
+      - |
+        {{- if .CMAKE_SETTINGS_DIR}}
+        cat <<EOF >> "{{.CMAKE_SETTINGS_DIR}}/boost.cmake"
+        set(Boost_ROOT "{{.INSTALL_PREFIX}}" CACHE PATH "Package root for boost.")
+        EOF
+        {{- end}}
```

Similar to the other changes, the reason for the original version/code was to use indentation to make it more readable. Iirc, indentation breaks the here-doc.
```suggestion
      - >-
        {{- if .CMAKE_SETTINGS_DIR}}
          echo "set(Boost_ROOT
            \"{{.INSTALL_PREFIX}}\"
            CACHE PATH
            \"Package root for boost.\"
          )" >> "{{.CMAKE_SETTINGS_DIR}}/boost.cmake"
        {{- end}}
```